### PR TITLE
[WFLY-16836] Drop source transformation of the EE security subsystem and remove the Jakarta EE 8 variant.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.redhat.com/browse/WFLY-16836

Hopefully this has captured all of the moving pieces to move from the Jakarta EE 8 variant.

This does also include the commit from https://github.com/wildfly/wildfly/pull/15950 as otherwise this sequence of changes will become hard to coordinate.

### CODE REVIEW

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml [hunk 1]
```diff
@@ -28,6 +28,7 @@
     </properties>
 
     <resources>
+        <artifact name="${org.glassfish.soteria:soteria.spi.bean.decorator.weld}"/>
         <artifact name="${org.glassfish.soteria:jakarta.security.enterprise}"/>
```
Comment (rhusar) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml:32]: I would keep these ordered alphabetically.

File: ee-security/pom.xml [hunk 2]
```diff
@@ -54,11 +54,15 @@
...
 
+        <!-- Jakarta-namespace specific deps -->
+
         <dependency>
             <groupId>${project.groupId}</groupId>
-            <artifactId>wildfly-ee</artifactId>
+            <artifactId>wildfly-ee-jakarta</artifactId>
         </dependency>
 
+        <!-- Other deps consistent with the javax.* ee-security module -->
```
Comment (rhusar) [inline ee-security/pom.xml:64]: This comment is kinda out of date now ;-)

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml (hunk 1 above)
Comment (darranl) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml:32]: @rhusar The reason I put this first was to ensure this decorator was picked up first.

File: ee-security/pom.xml (hunk 2 above)
Comment (darranl) [reply ee-security/pom.xml]: Removed

**Pull request merged**